File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EInverseOnSurfaceDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EInverseOnSurfaceDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd8
            AppTheme.STANDARD -> R.color.cstd8
            AppTheme.BLUE -> R.color.cbd8
            AppTheme.GREEN -> R.color.cgd8
            AppTheme.RED -> R.color.crd8
            AppTheme.NOTHING -> R.color.cnod8
        }
    }
}

File: lib/layer2_core/dark/e_inverse_on_surface_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EInverseOnSurfaceDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd8;
      case 'standard': return cstd8;
      case 'blue': return cbd8;
      case 'green': return cgd8;
      case 'red': return crd8;
      case 'nothing': return cnod8;
      default: return cstd8;
    }
  }
}
